packages/teleport/src/lib/tdp/codec.ts

```diff
@@ -389,6 +397,31 @@ export default class Codec {
     return buffer;
   }
 
+  // | message type (11) | completion_id uint32 | directory_id uint32 | name_length uint32 | name []byte |
+  encodeSharedDirectoryAnnounce(
```

Comment:
As discussed offline, it would be nice if we could add a test for this but as this is still in flux we can just make sure to revisit after the api's have stabilized.